### ISSUE TITLE
Significant trailing zeroes for nutrients are not kept

### ISSUE THREAD

**Issue by stephanegigandet** [openfoodfacts/openfoodfacts-server#3487]
While investigating bug #3483 , I realized that we are not keeping trailing 0s in nutrients sent by users or the API. e.g. is a nutrient value is 2.50, we convert the value to a number and it becomes 2.5.

We could try to preserve trailing 0s, but it may add a lot of complexity

**Comment by aleene** [comment]
I would prefer to keep the trailing zero's, as it is an indication of accuracy.

**Comment by stephanegigandet** [comment]
Yes, I would prefer too. It makes things quite complex though, as the values can be sent with one unit, but then converted / displayed in another. Or maybe we just store a new field carbohydrates_zeroes=2 if we are sent 2.00 g or 2000.00 mg.

**Comment by aleene** [comment]
In that way it would be dependent on the unit that has been used. Maybe it is better to define a field with the number of (significant?) digits, independent on the location of the comma. So 2.00 g would be 3, 2000.00 mg would be 6.

**Comment by VaiTon** [comment]
What about just having two decimal positions? 
3.956 -> 3.96
1.444 -> 1.45
2 -> 2.00
1.2 -> 1.20
130 -> 130.00

**Comment by teolemon** [comment]
@stephanegigandet has it been deployed ?

**Comment by teolemon** [comment]
Hasn't the nutrition refactor fixed this ?

**Comment by teolemon** [comment]
+2020 PR

**Comment by Freso** [comment]
FWIW, if significant trailing zeroes are stored/kept now, they’re not being shown to the user when editing… but that’s probably a separate issue to this one.
